core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -82,6 +75,33 @@ def start_navigator(self) -> None:
     def start_serial(self, device: str) -> None:
         self.start_mavlink_manager(Endpoint("Master", self.settings.app_name, "serial", device, 115200, protected=True))
 
+    def run_with_sitl(self, vehicle: str = "vectored") -> None:
+        firmware = os.path.join(self.settings.firmware_path, "sitl")
+        if not os.path.exists(firmware):
+            temporary_file = self.firmware_download.download(Vehicle.Sub, Platform.SITL)
```

Comment:
we shouldn't have to download it everytime. This is less of an issue for SITL, we should have firmware available for flashing when there's no internet connectivity, as we do in the current companion.

This is scope creep, though, so we can just open an issue.

Reply:
Didn't get it.
Currently we download the firmware only if the binary does not exist already (`if not os.path.exists(firmware)`).

Reply:
I missed that somehow. my bad. the path is in the settings folder, which means it is in the host filesystem, correct? nice

Reply:
No prob.
Yes, that's the default behavior. It's usually under `/root/.config/ardupilot-manager/`